privacyidea/static/components/token/controllers/tokenControllers.js

```diff
@@ -524,29 +525,28 @@ myApp.controller("tokenEnrollController", function ($scope, TokenFactory,
             radius.server, radius.secret...
            are stored in systemDefault and $scope.form
          */
-        var systemDefault = data.result.value;
+        $scope.systemDefault = data.result.value;
```

Comment:
It the user is not admin, he will not be able to load systemconfig. (see above).
What happens then? The ``$scope.systemDefault`` might not be initialized and it could break in line 276 or 279?
The ``$scope.form`` did exist for both ROLE.ADMIN and ROLE.USER.
Maybe we have to initialize ``$scope.systemDefault``. Is the "systemDefaul" actualy that, what we want?

Reply:
Well actually the user is able to load the (reduced) system config which contains the HOTP and TOTP parameter.
The `$scope.systemDefault` is initialized when the `tokenEnrollController` is instantiated.
The problem with `$scope.form` is, that everything that is put there will be send to privacyIDEA in the token enrollment call. And if we have i.e. a policy which enforces the HOTP hashlib to be `sha512`, but a system config where `hotp.hashlib = sha1` we would have both hashlibs in the token info (`hashlib: sha512, hotp.hashlib: sha1`) while the actual hashlib is `sha1`.
We can move this into #1839